### PR TITLE
[Document] [seatunnel-connector-V1-flink-kafka] The parameter matches the document

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
…document

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

correlation #3289

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/sink/KafkaSink.java [hunk 1]
```diff
@@ -67,12 +67,12 @@ public Config getConfig() {
 
     @Override
     public CheckResult checkConfig() {
-        return CheckConfigUtil.checkAllExists(config, "topics");
+        return CheckConfigUtil.checkAllExists(config, "topic");
     }
 
     @Override
     public void prepare(FlinkEnvironment env) {
-        topic = config.getString("topics");
```
Comment (CalvinKirs) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/sink/KafkaSink.java]: Considering the issue of compatibility, it is best to discuss it first

Comment (CalvinKirs) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/sink/KafkaSink.java]: For Flink engine, we should used `topic`

Comment (Carl-Zhou-CN) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-kafka/src/main/java/org/apache/seatunnel/flink/kafka/sink/KafkaSink.java]: OK

File: docs/en/connector/sink/Kafka.mdx [hunk 2]
```diff
@@ -17,48 +20,114 @@ Engine Supported and plugin name
...
+    topic = "seatunnel"
+    producer.bootstrap.servers = "localhost:9092"
+}
+```
+
+</TabItem>
+<TabItem value="flink">
+
+```bash
+    kafka {
```
Comment (hailin0) [inline docs/en/connector/sink/Kafka.mdx]:
```suggestion
kafka {
```

File: docs/en/connector/sink/Kafka.mdx [hunk 3]
```diff
@@ -17,48 +20,114 @@ Engine Supported and plugin name
...
+    defaultValue="spark"
+    values={[
+        {label: 'Spark', value: 'spark'},
+        {label: 'Flink', value: 'flink'},
+    ]}>
+<TabItem value="spark">
+
 ```bash
-kafka {
+    kafka {
```
Comment (hailin0) [inline docs/en/connector/sink/Kafka.mdx]:
```suggestion
kafka {
```

Comment (Carl-Zhou-CN) [reply docs/en/connector/sink/Kafka.mdx]: OK

**Review by hailin0 — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**